services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankealueEntity.kt

```diff
@@ -50,6 +56,15 @@ class HankealueEntity(
         orphanRemoval = true
     )
     var tormaystarkasteluTulos: TormaystarkasteluTulosEntity?,
+    @ElementCollection(fetch = FetchType.LAZY)
+    @CollectionTable(
+        name = "hankkeen_haittojenhallintasuunnitelma",
+        joinColumns = [JoinColumn(name = "hankealue_id", referencedColumnName = "id")]
+    )
+    @MapKeyColumn(name = "tyyppi")
+    @Column(name = "sisalto")
+    @MapKeyEnumerated(EnumType.STRING)
+    var haittojenhallintasuunnitelma: Map<Haittojenhallintatyyppi, String>? = null
```

Comment:
Should this be a mutable map so the values can be updated?

Reply:
It is now mutable. Even though the update always clears the map before inserting the new values.